packages/inter-protocol/src/auction/auctioneer.js

```diff
@@ -0,0 +1,284 @@
+      const { collateralSeat, currencySeat } = book.getSeats();
+
+      const depositsForBrand = deposits.get(brand);
+      if (depositsForBrand.length === 1) {
+        // send it all to the one
+        const liqSeat = depositsForBrand[0].seat;
+
+        atomicRearrange(
+          zcf,
+          harden([
+            [collateralSeat, liqSeat, collateralSeat.getCurrentAllocation()],
+            [currencySeat, liqSeat, currencySeat.getCurrentAllocation()],
+          ]),
+        );
+        liqSeat.exit();
+        deposits.set(brand, []);
+      } else if (depositsForBrand.length > 1) {
+        const totCollDeposited = depositsForBrand.reduce((prev, { amount }) => {
+          return AmountMath.add(prev, amount);
+        }, AmountMath.makeEmpty(brand));
+
+        const collatRaise = collateralSeat.getCurrentAllocation().Collateral;
+        const currencyRaise = currencySeat.getCurrentAllocation().Currency;
+
+        const collShare = makeRatioFromAmounts(collatRaise, totCollDeposited);
+        const currShare = makeRatioFromAmounts(currencyRaise, totCollDeposited);
+        /** @type {import('@agoric/zoe/src/contractSupport/atomicTransfer.js').TransferPart[]} */
+        const transfers = [];
+        let currencyLeft = currencyRaise;
+        let collateralLeft = collatRaise;
+
+        // each depositor gets as share that equals their amount deposited
+        // divided by the total deposited multplied by the currency and
+        // collateral being distributed.
+        for (const { seat, amount } of deposits.get(brand).values()) {
+          const currPortion = floorMultiplyBy(amount, currShare);
+          currencyLeft = AmountMath.subtract(currencyLeft, currPortion);
+          const collPortion = floorMultiplyBy(amount, collShare);
+          collateralLeft = AmountMath.subtract(collateralLeft, collPortion);
+          transfers.push([currencySeat, seat, { Currency: currPortion }]);
+          transfers.push([collateralSeat, seat, { Collateral: collPortion }]);
+        }
+
+        // TODO The leftovers should go to the reserve, and should be visible.
+        const keyword = brandToKeyword.get(brand);
+        transfers.push([
+          currencySeat,
+          reserveFunds,
+          { Currency: currencyLeft },
+        ]);
+        transfers.push([
+          collateralSeat,
+          reserveFunds,
+          { Collateral: collateralLeft },
+          { [keyword]: collateralLeft },
+        ]);
+        atomicRearrange(zcf, harden(transfers));
+
+        for (const { seat } of depositsForBrand) {
+          seat.exit();
+        }
+      }
+    }
+  };
+
+  const { publicMixin, creatorMixin, makeFarGovernorFacet, params } =
+    await handleParamGovernance(
+      zcf,
+      privateArgs.initialPoserInvitation,
+      // @ts-expect-error XXX How to type this?
+      auctioneerParamTypes,
+      privateArgs.storageNode,
+      privateArgs.marshaller,
+    );
+
+  const tradeEveryBook = () => {
```

Comment:
re: the book metaphor comment above, this is enumerating the `auctionBook` but and not the two sub "books"

Reply:
In my mental model, the Auctioneer knows that it has a book per collateral type, but doesn't think about how those books represent orders. The AuctionBooks may be confused, but it shouldn't show through here.